neurodsp/filt/checks.py

```diff
@@ -133,26 +133,103 @@ def check_filter_properties(b_vals, a_vals, fs, pass_type, f_range,
     >>> passes = check_filter_properties(filter_coefs, 1, fs, pass_type, f_range)
     Transition bandwidth is 0.5 Hz.
     Pass/stop bandwidth is 24.0 Hz.
+    """
+
+    # Import utility functions inside function to avoid circular imports
+    from neurodsp.filt.utils import (compute_frequency_response,
+                                     compute_pass_band, compute_transition_band)
+
+    # Initialize variable to keep track if all checks pass
+    passes = True
+
+    # Compute the frequency response
+    f_db, db = compute_frequency_response(b_vals, a_vals, fs)
+
+    # Check that frequency response goes below transition level (has significant attenuation)
+    if np.min(db) >= transitions[0]:
+        passes = False
+        warn('The filter attenuation never goes below {} dB.'\
+             'Increase filter length.'.format(transitions[0]))
+        # If there is no attenuation, cannot calculate bands, so return here
+        return passes
 
+    # Check that both sides of a bandpass have significant attenuation
+    if pass_type == 'bandpass':
+        if db[0] >= transitions[0] or db[-1] >= transitions[0]:
+            passes = False
+            warn('The low or high frequency stopband never gets attenuated by'\
+                 'more than {} dB. Increase filter length.'.format(abs(transitions[0])))
+
+    # Compute pass & transition bandwidth
+    pass_bw = compute_pass_band(fs, pass_type, f_range)
+    transition_bw = compute_transition_band(f_db, db, transitions[0], transitions[1])
+
+    # Raise warning if transition bandwidth is too high
+    if transition_bw > pass_bw:
+        passes = False
+        warn('Transition bandwidth is  {:.1f}  Hz. This is greater than the desired'\
+             'pass/stop bandwidth of  {:.1f} Hz'.format(transition_bw, pass_bw))
+
+    # Print out transition bandwidth and pass bandwidth to the user
+    if verbose:
+        print('Transition bandwidth is {:.1f} Hz.'.format(transition_bw))
+        print('Pass/stop bandwidth is {:.1f} Hz.'.format(pass_bw))
+
+    return passes
+
+def sos_check_filter_properties(sos, fs, pass_type, f_range,
```

Comment:
Tests needed for this func and `sos_compute_frequency_response` need unit tests.